rn/Teacher/src/modules/submissions/list/get-submissions-props.js

```diff
@@ -142,45 +128,79 @@ export function pendingProp (assignmentContent?: AssignmentContentState, courseC
   return assignmentContent.submissions.pending > 0 || courseContent.enrollments.pending > 0
 }
 
+export function getGroup (groups: GroupsState, submission: SubmissionWithHistory, categoryID: ?string): ?Group {
```

Comment:
If a submission does not have a grade or an actual submission it will not have a group attached to the submission. So if it's there we get the group from entities, but if not then we will look up the group in entities based on the category id and if the user is in the group.